### PR TITLE
[astro] Add getTotalRadiation to AstroActions 

### PR DESCRIPTION
**Pull request by sislakd**
This is enhancement of Astro binding. There is added getTotalRadiation to AstroActions allowing the user to get total radiation of the sun at the requested instant. The documentation of the Astro binding is updated as well. The proposed enhancement is tested and works also in the previous version 3.4.2.

### CODE REVIEW

**Comment by sislakd** [comment]
I would like to add Enhancement label, but it seems I don't have permissions to do that.

**Review by clinique — APPROVED** [review]

File: bundles/org.openhab.binding.astro/README.md [hunk 1]
```diff
@@ -265,6 +265,15 @@ Example :
  logInfo("AstroActions", "{} will be positioned at elevation {} - azimuth {}",sunEvent, elevation.toString,azimuth.toString)
 ```
 
+### getTotalRadiation(timeStamp)
+Retrieves the total radiation(QuantityType) of the sun at the requested instant.
+Thing method only applies to Sun thing type.
+
+```java
+ val totalRadiation = sunActions.getTotalRadiation(ZonedDateTime.now)
+ logInfo("AstroActions", "Currently, the total sun radiation is {}",   totalRadiation.toString)
```
Comment (jlaur) [inline bundles/org.openhab.binding.astro/README.md:274]:
```suggestion
 logInfo("AstroActions", "Currently, the total sun radiation is {}", totalRadiation.toString)
```

File: bundles/org.openhab.binding.astro/src/main/java/org/openhab/binding/astro/internal/action/AstroActions.java [hunk 2]
```diff
@@ -86,6 +88,25 @@ public void setThingHandler(@Nullable ThingHandler handler) {
...
 
+    @RuleAction(label = "get the total sun radiation", description = "Get the total sun radiation for a given time.")
+    public @Nullable @ActionOutput(name = "getTotalRadiation", label = "Total Radiation", type = "org.openhab.core.library.types.QuantityType<org.openhab.core.library.dimension.Intensity>") QuantityType<Intensity> getTotalRadiation(
+            @ActionInput(name = "date", label = "Date", required = false, description = "Considered date") @Nullable ZonedDateTime date) {
+        logger.debug("Astro action 'getTotalRadiation' called");
+        AstroThingHandler theHandler = this.handler;
+        if (theHandler != null) {
+            if (theHandler instanceof SunHandler) {
+                SunHandler handler = (SunHandler) theHandler;
+                Radiation radiation = handler.getRadiationAt(date != null ? date : ZonedDateTime.now());
```
Comment (jlaur) [inline bundles/org.openhab.binding.astro/src/main/java/org/openhab/binding/astro/internal/action/AstroActions.java:99]:
```suggestion
            if (theHandler instanceof SunHandler sunHandler) {
                Radiation radiation = sunHandler.getRadiationAt(date != null ? date : ZonedDateTime.now());
```

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks! Only some minor comments.

**Comment by sislakd** [comment]
@jlaur Thanks for the feedback. I've reflected your changes.

**Review by jlaur — APPROVED** [review]
LGTM

File: bundles/org.openhab.binding.astro/src/main/java/org/openhab/binding/astro/internal/action/AstroActions.java (hunk 2 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.astro/src/main/java/org/openhab/binding/astro/internal/action/AstroActions.java]: As this is a java 17 feature this suggested change is not 3.4.x compatible as stated in the start post. Not a big deal though.

Comment (jlaur) [reply bundles/org.openhab.binding.astro/src/main/java/org/openhab/binding/astro/internal/action/AstroActions.java]: Correct, as an enhancement this is not being considered for 3.4.x anyway.

**Pull request merged**